src/containers/Explorer/FilterNav.js

```diff
@@ -33,13 +41,64 @@ const FilterNavComponent = props => {
     categories,
     categoryFilters,
     addCategoryFilter,
-    removeCategoryFilter
+    removeCategoryFilter,
+    history,
+    location,
+    batch
   } = props;
 
+  const rootLocationParams = location.search || '?bountyStage=active';
+
+  const toggleStageFilterAction = stage => {
+    const queryParams =
+      toggleFromParam(rootLocationParams, 'bountyStage', stage) ||
+      '?bountyStage=';
+    history.push(location.pathname + queryParams);
+    toggleStageFilter(stage);
+  };
+
+  const toggleDifficultyFilterAction = difficulty => {
+    history.push(
+      location.pathname +
+        toggleFromParam(rootLocationParams, 'difficulty', difficulty)
+    );
+    toggleDifficultyFilter(difficulty);
+  };
+
+  const addCategoryFilterAction = category => {
+    history.push(
+      location.pathname + pushToParam(rootLocationParams, 'category', category)
+    );
+    addCategoryFilter(category);
+  };
+
+  const removeCategoryFilterAction = category => {
+    history.push(
+      location.pathname +
+        removeFromParam(rootLocationParams, 'category', category)
+    );
+    removeCategoryFilter(category);
+  };
+
+  const setSearchAction = throttle(300, searchValue => {
+    history.push(
+      location.pathname + setParam(rootLocationParams, 'search', searchValue)
+    );
+    setSearch(searchValue);
+  });
+
+  const resetFilterAction = () => {
+    history.push(location.pathname);
+    batch(true);
```

Comment:
example of batching. We can give it a different name if we'd like... Alternatively - instead of this batching pattern, we could just duplicate and not make it so any of the filter actions trigger an API call. We could require an explicity loadBounties every time you want to refresh the list. However, I tend to like this pattern better. Let me know your thoughts @c-o-l-o-r